deployer/hub.py

```diff
@@ -140,6 +142,46 @@ def auth_kubeconfig(self):
             os.environ['KUBECONFIG'] = decrypted_key_path
             yield
 
+    def auth_aws(self):
+        config = self.spec['aws']
+        key_path = config['key']
+        cluster_type = config['clusterType']
+        cluster_name = config['clusterName']
+        region = config['region']
+
+        with tempfile.NamedTemporaryFile() as kubeconfig:
+            orig_kubeconfig = os.environ.get('KUBECONFIG', None)
+            orig_access_key_id = os.environ.get('AWS_ACCESS_KEY_ID', None)
+            orig_secret_access_key = os.environ.get('AWS_SECRET_ACCESS_KEY', None)
+            try:
+                os.environ['KUBECONFIG'] = kubeconfig.name
+                with decrypt_file(key_path) as decrypted_key_path:
+
+                    decrypted_key_abspath = os.path.abspath(decrypted_key_path)
+                    if not os.path.isfile(decrypted_key_abspath):
+                        raise FileNotFoundError(
+                            'The decrypted key file does not exist')
+                    with open(decrypted_key_abspath) as f:
+                        creds = json.load(f)
+
+                    os.environ['AWS_ACCESS_KEY_ID'] = creds["AccessKey"]['AccessKeyId']
+                    os.environ['AWS_SECRET_ACCESS_KEY'] = creds["AccessKey"]['SecretAccessKey']
+
+                subprocess.check_call([
+                    'aws', 'eks', 'update-kubeconfig',
+                    f'--name={cluster_name}',
+                    f'--region={region}'
+                ])
+
+                yield
+            finally:
+                if orig_kubeconfig is not None:
+                    os.environ['KUBECONFIG'] = orig_kubeconfig
+                if orig_access_key_id is not None:
+                    os.environ['AWS_ACCESS_KEY_ID'] = orig_access_key_id
+                if orig_kubeconfig is not None:
+                    os.environ['AWS_SECRET_ACCESS_KEY'] = orig_secret_access_key
```

Comment:
In AWS, sometimes you have an account with permissions to act as a certain role. We don't have support for that currently, so if you manage a cluster you need to make every user individually a k8s admin, rather than define what it means to be a k8s admin once and let everyone be able to act as such role.

The JMTE cluster is setup to have a k8s role and a ecr role (where we push an image).

Reply:
Yep, as I said in another reply:

> I tried this approach first because it is the most straightforward and because none of our current docs or practices in AWS land includes AWS roles... yet.

I would be happy to explore using roles in the near future.